src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceGlyphManager.cs

```diff
@@ -0,0 +1,186 @@
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.VisualStudio.LanguageServices.Implementation.InheritanceMargin.MarginGlyph;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Formatting;
+using Microsoft.VisualStudio.Utilities;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.InheritanceMargin
+{
+    /// <summary>
+    /// Manager controls all the glyph of Inheritance Margin
+    /// </summary>
+    internal class InheritanceGlyphManager
+    {
+        private readonly IWpfTextView _textView;
+        private readonly IThreadingContext _threadingContext;
+        private readonly IStreamingFindUsagesPresenter _streamingFindUsagesPresenter;
+        private readonly ClassificationTypeMap _classificationTypeMap;
+        private readonly IClassificationFormatMap _classificationFormatMap;
+        private readonly IUIThreadOperationExecutor _operationExecutor;
+        private readonly IEditorFormatMap _editorFormatMap;
+        private Dictionary<InheritanceMarginGlyph, SnapshotSpan> _glyphToTaggedSpan;
+        private readonly Canvas _glyphsContainer;
+
+        // We want to our glyphs to have the same background color as the glyphs in GlyphMargin
+        private const string GlyphMarginName = "Indicator Margin";
+
+        public InheritanceGlyphManager(
+            IWpfTextView textView,
+            IThreadingContext threadingContext,
+            IStreamingFindUsagesPresenter streamingFindUsagesPresenter,
+            ClassificationTypeMap classificationTypeMap,
+            IClassificationFormatMap classificationFormatMap,
+            IUIThreadOperationExecutor operationExecutor,
+            IEditorFormatMap editorFormatMap,
+            Canvas canvas)
+        {
+            _textView = textView;
+            _threadingContext = threadingContext;
+            _streamingFindUsagesPresenter = streamingFindUsagesPresenter;
+            _classificationTypeMap = classificationTypeMap;
+            _classificationFormatMap = classificationFormatMap;
+            _operationExecutor = operationExecutor;
+            _editorFormatMap = editorFormatMap;
+            _glyphsContainer = canvas;
+            _editorFormatMap.FormatMappingChanged += FormatMappingChanged;
+
+            _glyphToTaggedSpan = new Dictionary<InheritanceMarginGlyph, SnapshotSpan>();
+            _textView.Closed += OnClosed;
+            UpdateBackgroundColor();
+        }
+
+        public void AddGlyph(InheritanceMarginTag tag, SnapshotSpan span)
+        {
+            var lines = _textView.TextViewLines;
+            if (GetStartingLine(lines, span) is IWpfTextViewLine line)
+            {
+                var glyph = CreateNewGlyph(tag);
+                glyph.Height = 17;
+                glyph.Width = 17;
```

Comment:
is it ok to hardcode tehse numbers like this?

Reply:
This is the same value hard-coded by the editor implementation
https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=%2Fsrc%2FEditor%2FText%2FImpl%2FGlyphMargin%2FGlyphMargin.cs&version=GBmain&line=54&lineEnd=55&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
yes, it's the same as the editor implementation. I will extract this value to a const field